src/extlibs/hrvo/hrvo_agent.cpp

```diff
@@ -146,33 +185,31 @@ void HRVOAgent::computeVelocityObstacles()
         velocity_obstacles_.push_back(velocity_obstacle);
     }
 
-    // Create Velocity Obstacles for nearby static obstacles
-    Point agent_position_point(getPosition());
     Circle circle_rep_of_agent(agent_position_point, radius_);
-    Segment path(agent_position_point, Point(current_destination));
-    for (const auto &obstacle : static_obstacles)
-    {
-        double dist_agent_to_obstacle = obstacle->distance(agent_position_point);
+    Segment path_segment(agent_position_point, current_destination);
 
-        // Set of heuristics to minimize the amount of velocity obstacles
-        if ((obstacle->intersects(path) ||
-             dist_agent_to_obstacle < 2 * ROBOT_MAX_RADIUS_METERS) &&
-            !obstacle->contains(agent_position_point))
+    // Set of heuristics to minimize the amount of velocity obstacles
+    for (const auto &obstacle : dynamic_obstacles)
+    {
+        // Add obstacle if the current path goes through it, or if we're very close to it
+        if (obstacle->intersects(path_segment) ||
+            obstacle->distance(agent_position_point) < 4 * ROBOT_MAX_RADIUS_METERS)
         {
             VelocityObstacle velocity_obstacle =
                 obstacle->generateVelocityObstacle(circle_rep_of_agent, Vector());
             velocity_obstacles_.push_back(velocity_obstacle);
         }
     }
 
-    // The conditions for creating a velocity obstacle for the ball are different,
-    // since the ball is a dynamic obstacle (not considered by the path planner)
-    // and `generateVelocityObstacle` can create valid velocity obstacles for agents
-    // contained in a circle.
-    if (ball_obstacle.has_value())
+    // Create Velocity Obstacles for nearby static obstacles
+    for (const auto &obstacle : static_obstacles)
     {
-        auto obstacle = ball_obstacle.value();
-        if (obstacle->intersects(path))
+        double dist_agent_to_obstacle = obstacle->distance(agent_position_point);
+
+        // Set of heuristics to minimize the amount of velocity obstacles
+        if ((obstacle->intersects(path_segment) ||
+             dist_agent_to_obstacle < 2 * ROBOT_MAX_RADIUS_METERS) &&
+            !obstacle->contains(agent_position_point))
```

Comment:
add more brackets here to make it clearer to the reader which conditions the || and && apply to